### DISCUSSION TITLE
`gradlew wrapper` ignores GRADLE_HOME environment variable

### DISCUSSION THREAD

**Discussion by TWiStErRob** [renovatebot/renovate#17639] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.190.4

### If you're self-hosting Renovate, select which platform you are using.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

A system can have a [GRADLE_USER_HOME](https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_environment_variables) environment variable set, which guides `gradlew` where to put the wrappers and the caches.

Renovate seems to ignore this variable and tries to use the default `~/.gradle` folder. Based on a quick look I think this is because the environment variables are not inherited here from the Node.js running Renovate:
https://github.com/renovatebot/renovate/blob/b8004fe61ada7ffbe0f3101dc39127e37be4ee81/lib/modules/manager/gradle-wrapper/artifacts.ts#L92-L107
more specifically somewhere here:
https://github.com/renovatebot/renovate/blob/b8004fe61ada7ffbe0f3101dc39127e37be4ee81/lib/util/exec/index.ts#L34-L39

Another important environment variable is GRADLE_OPTS and JAVA_TOOL_OPTIONS, both of which can be used to specify JVM arguments. This can be important on a shared server where spawning a Java process might take up 1/4 of the memory (this is the default 🤦‍♂️), but on a server with 64GB of RAM this means 16GB RAM, `gradlew` spawns a JVM process, which means to replace a 60kB-ish jar file, Renovate uses 16GB RAM... this can be worse if the repository contains multiple gradle-wrapper.properties, because each file triggers a `gradlew wrapper` execution.

There are more variables which can affect the JVM like this, so a full propagation of env vars would be ideal. Here's a few: https://stackoverflow.com/questions/28327620/difference-between-java-options-java-tool-options-and-java-opts

### Relevant debug logs

The logs below show a failure, because `C:\\Users\\TWiStEr\\.gradle` folder is not writable, `GRADLE_USER_HOME` is.

<details><summary>Logs</summary>

```
 WARN: Error executing gradle wrapper update command. It can be not a critical one though. (repository=TWiStErRob/net.twisterrob.gradle, branch=renovate/gradle-7.x)
       "err": {
         "name": "ExecError",
         "cmd": "C:\\WINDOWS\\system32\\cmd.exe /d /s /c \"gradlew.bat wrapper --gradle-distribution-url https://services.gradle.org/distributions/gradle-7.5.1-all.zip\"",
         "stderr": "Exception in thread \"main\" java.lang.RuntimeException: Could not create parent directory for lock file C:\\Users\\TWiStEr\\.gradle\\wrapper\\dists\\gradle-7.5.1-all\\1ehga6e77gqps5uk2kc5kf1vc\\gradle-7.5.1-all.zip.lck\r\n\tat
org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:43)\r\n\tat org.gradle.wrapper.Install.createDist(Install.java:48)\r\n\tat org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:107)\r\n\tat org.gradle.
wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)\r\n",
         "stdout": "",
         "options": {
           "cwd": "C:/Users/TWiStEr/AppData/Local/Temp/renovate/repos/github/TWiStErRob/net.twisterrob.gradle/docs/debug/agp333-gradle4103",
           "encoding": "utf-8",
           "env": {
             "GRADLE_OPTS": "-Xmx512M",
             "HOME": "C:\\Users\\TWiStEr",
             "PATH": "..."
           },
           "maxBuffer": 10485760,
           "timeout": 900000
         },
         "exitCode": 1,
         "message": "Command failed: gradlew.bat wrapper --gradle-distribution-url https://services.gradle.org/distributions/gradle-7.5.1-all.zip\nException in thread \"main\" java.lang.RuntimeException: Could not create parent directory for lock f
ile C:\\Users\\TWiStEr\\.gradle\\wrapper\\dists\\gradle-7.5.1-all\\1ehga6e77gqps5uk2kc5kf1vc\\gradle-7.5.1-all.zip.lck\r\n\tat org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:43)\r\n\tat org.gradle.wrapper.Insta
ll.createDist(Install.java:48)\r\n\tat org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:107)\r\n\tat org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)\r\n",
         "stack": "ExecError: Command failed: gradlew.bat wrapper --gradle-distribution-url https://services.gradle.org/distributions/gradle-7.5.1-all.zip\nException in thread \"main\" java.lang.RuntimeException: Could not create parent directory f
or lock file C:\\Users\\TWiStEr\\.gradle\\wrapper\\dists\\gradle-7.5.1-all\\1ehga6e77gqps5uk2kc5kf1vc\\gradle-7.5.1-all.zip.lck\r\n\tat org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:43)\r\n\tat org.gradle.wrap
per.Install.createDist(Install.java:48)\r\n\tat org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:107)\r\n\tat org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)\r\n\n    at ChildProcess.<anonymous> (P:\\projects\\co
ntrib\\github-twisterrob-renovate\\node_modules\\renovate\\dist\\util\\exec\\common.js:87:24)\n    at ChildProcess.emit (node:events:538:35)\n    at ChildProcess.emit (node:domain:475:12)\n    at Process.ChildProcess._handle.onexit (node:internal/c
hild_process:291:12)"
       }
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
this isn't a bug. renovate will trim child ENV by default.
you can configure custom ENV bars in renovate self-hosted config or enable expose all ENV.

**Reply by TWiStErRob** [reply]
but these are common env vars for any Java commands, even on your hosting envs you might be consuming way more RAM than necessary. These apply to anyone running `gradlew`.

**Reply by TWiStErRob** [reply]
Note: here are the links to the configs you mentioned:
https://docs.renovatebot.com/self-hosted-configuration/#customenvvariables
https://docs.renovatebot.com/self-hosted-configuration/#exposeallenv

The second one will solve my problem, but it's still a strange setup to not handle it by default, isn't it?

**Comment by TWiStErRob** [comment]
I added the variables mentioned above to
https://github.com/renovatebot/renovate/blob/b8004fe61ada7ffbe0f3101dc39127e37be4ee81/lib/util/exec/env.ts#L3-L17
and I can confirm they're picked up. Still fails for some reason, but I can't read out yet why.

**Reply by TWiStErRob** [reply]
Ah, I also need to expose `ANDROID_SDK_ROOT`, because the project running the `wrapper` task uses Android SDK.

Does this mean that Renovate won't be able to update Gradle on Android projects?
